### PR TITLE
Remove Config Defaults from Scala code

### LINKED ISSUE #2326: Remove defaults for config parameters in code
To see what could be configured and what are the defaults _easily_, we should have all config options with their defaults in the `application.conf`. To ensure this, we can simply remove the defaults in the code and fail if something is not set.

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/2326/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by fm3**
### Steps to test:
- remove a required config value from application.conf
- wK startup should fail with readable exception
- with the application.conf in the repo, it should run
- also test that [standalone datastores](https://github.com/scalableminds/webknossos/wiki/Set-up-a-standalone-datastore-locally) still work

### TODO/Decisions
- [x] also require config keys used by frameworks, not used by us? → No, we decided that’s redundant and doesn’t help, as we cannot make guarantees about its completeless
- [x] figure out what changes are necessary in kube/migration
- [x] decide wether to name the nested objects all-lower-case to match config keys → No, keep it as it is

### Issues:
- fixes #2326 

------
- ~~[ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- [x] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

**Comment by fm3** [comment]
note that the config objects are implemented in two different ways. For the datastore it’s already done in the dependcy-injection kind of way, while for webKnossos it is still a scala object for compatibility (see related #3059 )

File: app/controllers/OrganizationController.scala [hunk 1]
```diff
@@ -22,11 +23,7 @@ class OrganizationController @Inject()(val messagesApi: MessagesApi) extends Con
...
   }
 
-  def getOperatorData = Action.async { implicit request =>
-    for {
-      data <- Play.configuration.getString("operatorData").toFox
-    } yield {
-      Ok(Json.toJson(data))
-    }
+  def getOperatorData = Action { implicit request =>
+      Ok(Json.toJson(WkConf.operatorData))
```
Comment (jstriebel) [inline app/controllers/OrganizationController.scala]: :rofl: nice catch

File: app/utils/WkConf.scala [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  object DataStore {
+    val enabled = getBoolean("datastore.enabled")
+    val key = getString("datastore.key")
+  }
+
+  object User {
+    val cacheTimeoutInMinutes = getInt("user.cacheTimeoutInMinutes") minutes
+  }
+
+  object BrainTracing {
```
Comment (jstriebel) [inline app/utils/WkConf.scala]: let's stick to the capitalization of the key, so `Braintracing`

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/DataStoreConfig.scala [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  object DataStore {
+    val key = getString("datastore.key")
+    val name = getString("datastore.name")
+    object Oxalis {
+      val uri = getString("datastore.oxalis.uri")
+      val secured = getBoolean("datastore.oxalis.secured")
+      val pingIntervalMinutes = getInt("datastore.oxalis.pingIntervalMinutes") minutes
+    }
+    object FossilDB {
```
Comment (jstriebel) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/DataStoreConfig.scala]: `Fossildb`

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/DataStoreConfig.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        val enabled = getBoolean("braingames.binary.changeHandler.enabled")
+        val tickerInterval = getInt("braingames.binary.changeHandler.tickerInterval") minutes
+      }
+      val baseFolder = getString("braingames.binary.baseFolder")
+      val loadTimeout = getInt("braingames.binary.loadTimeout") seconds
+      val cacheMaxSize = getInt("braingames.binary.cacheMaxSize")
+    }
+  }
+
+  object DataStore {
```
Comment (jstriebel) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/DataStoreConfig.scala]: `Datastore`

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/DataSourceService.scala [hunk 5]
```diff
@@ -9,36 +6,35 @@ import java.nio.file.{AccessDeniedException, Path, Paths}
...
+                                   config: DataStoreConfig,
                                    dataSourceRepository: DataSourceRepository,
                                    val lifecycle: ApplicationLifecycle,
                                    @Named("webknossos-datastore") val system: ActorSystem
                                  ) extends IntervalScheduler with LazyLogging with FoxImplicits {
 
-  override protected lazy val enabled: Boolean = config.getBoolean("braingames.binary.changeHandler.enabled").getOrElse(true)
-  protected lazy val tickerInterval: FiniteDuration = config.getInt("braingames.binary.changeHandler.interval").getOrElse(10).minutes
+  override protected lazy val enabled: Boolean = config.Braingames.Binary.ChangeHandler.enabled
+  protected lazy val tickerInterval: FiniteDuration = config.Braingames.Binary.ChangeHandler.tickerInterval
```
Comment (jstriebel) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/DataSourceService.scala]: :+1:

**Comment by jstriebel** [comment]
PS: The `tools/extract_config_keys.sh` script (which should be removed with this PR), showed two more files with config usages: ` app/oxalis/mail/DefaultMails.scala` & `app/views/main.scala.html`

**Comment by fm3** [comment]
@jstriebel I adapted the spelling and replaced the remaining usages. I would not remove the script – maybe we could even run it in the CI and check its output to be empty?

**Pull request merged**